packages/api/src/campaign/entities/campaign-entity.factory.ts

```diff
@@ -0,0 +1,80 @@
+import { Block, BlockDataInterface, RootBlock } from "@comet/blocks-api";
+import { DocumentInterface, RootBlockDataScalar, RootBlockType } from "@comet/cms-api";
+import { Embedded, Entity, OptionalProps, PrimaryKey, Property } from "@mikro-orm/core";
+import { Type } from "@nestjs/common";
+import { Field, ID, Int, ObjectType } from "@nestjs/graphql";
+import { v4 } from "uuid";
+
+import { CampaignScopeInterface } from "../../types";
+import { SendingState } from "../sending-state.enum";
+
+export interface CampaignInterface {
+    [OptionalProps]?: "createdAt" | "updatedAt" | "sendingState";
+    id: string;
+    createdAt: Date;
+    scheduledAt?: Date;
+    title: string;
+    subject: string;
+    brevoId?: number;
+    updatedAt: Date;
+    content: BlockDataInterface;
+    scope: CampaignScopeInterface;
+}
+
+export class CampaignEntityFactory {
```

Comment:
Is CampaignEntity a good name or should it be more specific? Maybe:
NewsletterCampaignEntity or EmailCampaignEntity.

I think EmailCampaignEntity might be a better name, because that is how Brevo is calling it:
https://developers.brevo.com/reference/createemailcampaign-1